src/bin/facilitator-init.ts

```diff
@@ -30,8 +31,63 @@ import Repositories from '../repositories/Repositories';
 import SeedData from '../SeedData';
 import GatewayAddresses from '../Config/GatewayAddresses';
 
+/**
+ * It provides gateway addresses and origin chain id. It is to be used when mosaic
+ * config is provided.
+ * @param auxChainId Auxiliary chain id.
+ * @param mosaicConfigPath Path to mosaic config.
+ * @returns originchain id and gatewayaddresses object.
+ */
+function getFromMosaicConfig(
+  auxChainId: number,
+  mosaicConfigPath: string,
+): {
+    originChainId: string; gatewayAddresses: GatewayAddresses;
+  } {
+  const mosaicConfig = MosaicConfig.fromFile(mosaicConfigPath);
+  const auxChain = mosaicConfig.auxiliaryChains[auxChainId];
+  if (auxChain === null || auxChain === undefined) {
+    Logger.error(`auxchain id ${auxChainId} is not present in the mosaic config`);
+    process.exit(1);
+  }
+
+  return {
+    originChainId: mosaicConfig.originChain.chain,
+    gatewayAddresses: GatewayAddresses.fromMosaicConfig(mosaicConfig, auxChainId),
+  };
+}
+
+/**
+ * It provides gateway addresses and origin chain id. It is to be used when gateway
+ * config is provided.
+ * @param auxChainId Auxiliary chain id.
+ * @param gatewayConfigPath Path to gateway config.
+ * @returns originchain id and gatewayaddresses object.
+ */
+function getFromGatewayConfig(
+  auxChainId: number,
+  gatewayConfigPath: string,
+): {
+    originChainId: string; gatewayAddresses: GatewayAddresses;
+  } {
+  const gatewayConfig = GatewayConfig.fromFile(gatewayConfigPath);
+
+  if (auxChainId === gatewayConfig.auxChainId) {
+    Logger.error(`aux chain id present in gateway config is ${gatewayConfig.auxChainId}`
+        + `but ${auxChainId} is specified in the command`);
+    process.exit(1);
+  }
+
+  const gatewayAddresses = GatewayAddresses.fromGatewayConfig(gatewayConfig);
+  return {
+    originChainId: gatewayConfig.mosaicConfig.originChain.chain,
+    gatewayAddresses,
+  };
+}
+
 commander
   .option('-m, --mosaic-config <mosaic-config>', 'path to mosaic configuration')
+  .option('-q, --gateway-config <gateway-config>', 'path to gateway configuration')
```

Comment:
why did you decide on `q` as option? 
I think we should revisit all the options.

Reply:
@abhayks1  @puneet-khushwani-eth  what do you think?

Reply:
As discussed with @puneet-khushwani-eth and @abhayks1 , decided to go for **-g** for **--gateway-config**. So, **-n** is now used for **--origin-graph-rpc**.